sdk/appconfiguration/app-configuration/test/public/index.readonlytests.spec.ts

```diff
@@ -23,9 +23,9 @@ describe("AppConfigurationClient (set|clear)ReadOnly", () => {
   };
 
   beforeEach(async function (this: Context) {
-    recorder = startRecorder(this);
-    testConfigSetting.key = recorder.getUniqueName("readOnlyTests");
-    client = createAppConfigurationClientForTests() || this.skip();
+    recorder = await startRecorder(this);
+    testConfigSetting.key = recorder.variable("readOnlyTests", `readOnlyTests${Math.floor(Math.random() * 1000)}`);
+    client = createAppConfigurationClientForTests(recorder.configureClientOptions({})) || this.skip();
```

Comment:
Can you get rid of `this.skip()`?

Reply:
Out of scope to this PR, but while we are at it, we can do some cleaning 🙂

Reply:
If we get rid of `this.skip()`, we would have to do explicit casting for client type. Do you want to make this change to all the test files too?

Reply:
I don't want explicit casting. Instead, fix the method `createAppConfigurationClientForTests` to always return a client.


https://github.com/Azure/azure-sdk-for-js/pull/23520/files#r996138222

Updating this should help